modules/core/src/main/scala/org/tessellation/infrastructure/rewards/Rewards.scala

```diff
@@ -59,8 +63,7 @@ object Rewards {
           .flatTap {
             case (remaining, _) =>
               Applicative[F].whenA(remaining =!= Amount(0L))(
-                Applicative[F].unit
-//                logger.error(s"Some rewards were not distributed {amount=${amount.show}, remainingAmount=${remaining.show}}")
+                logger.error(s"Some rewards were not distributed {amount=${amount.show}, remainingAmount=${remaining.show}}")
```

Comment:
If such case happens it means that there is an implementation error in distribution programs. The question is if we should return reward transactions then anyway.

Reply:
Hm, yeah I think you're right. Maybe we should throw here.